pkg/client/unversioned/pod_logs.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"strconv"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// PodLogsNamespacer has methods to work with podLog subresources in a namespace
+type PodLogsNamespacer interface {
+	PodLogs(namespace string) PodLogsInterface
+}
+
+// PodLogsInterface has methods to work with podLog subresources.
+type PodLogsInterface interface {
+	Get(name string, opts *api.PodLogOptions) (*Request, error)
```

Comment:
One argument is that broadening the PodsInterface over the long term is a
bad trend - pod logs are a natural API to be moved out of the pod API
group.  I was ambivalent - its slightly inconsistent in the short term, but
probably correct in the long term.

On Oct 27, 2015, at 6:38 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/client/unversioned/pod_logs.go
<https://github.com/kubernetes/kubernetes/pull/10707#discussion_r43197769>:

> +
> +import (
> +	"strconv"
> +	"time"
> +
> +	"k8s.io/kubernetes/pkg/api"
> +)
> +
> +// PodLogsNamespacer has methods to work with podLog subresources in a namespace
> +type PodLogsNamespacer interface {
> +	PodLogs(namespace string) PodLogsInterface
> +}
> +
> +// PodLogsInterface has methods to work with podLog subresources.
> +type PodLogsInterface interface {
> +	Get(name string, opts *api.PodLogOptions) (*Request, error)

I think I'd prefer a GetLogs method in the existing pod client?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/10707/files#r43197769>.